server/api/service/geocoder.js

```diff
@@ -23,35 +24,75 @@ const degreeToMeter = (lon, lat, dn, de) => {
 // Some parcels (Gushim) are malformed (usually in the Golan Heights) and should be ignored.
 const BAD_PARCELS = ['203000', '202011', '202000', '201000', '200000', '200001'];
 
-const gushHelkaToPolygon = async (gush, helka) => {
+const gushHelkaToPolygon = async (gush, helka) => {		
 	if (!gush || !helka) {
 		return;
 	}
+	
 	if (BAD_PARCELS[gush]) {
 		return;
 	}
+
+
+	// parse the raw string into a unique array of numbers 
+	// e.g -
+    //  - "123,123" => [123]
+	//  - "1-2" => [1,2]
+	//  - "helka 4 and 5" => [4,5]  
+	const helkaArr = [...new Set(helka.match(/\d+/gi))]
+
+
+	// filter the raw string into a single numeric value 
+	// e.g - "194,194" => 194
+	const filteredGush = (gush.match(/\d+/i) || [])[0]
+	
+	
+	console.log('filtered gush and helka', 'helkaArr', helkaArr, 'filteredGush', filteredGush)
 	try {
-		const res = await axios(
-			{
-				url: `https://open.govmap.gov.il/geoserver/opendata/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=2.0.0&TYPENAMES=opendata:PARCEL_ALL&TYPENAME=opendata:PARCEL_ALL&outputFormat=json&cql_FILTER=GUSH_NUM=${gush}AND%20PARCEL=${helka}&srsName=EPSG:4326`,
-				method: 'get',
-				timeout: 20000,
-			});
-		const geoJsonRes = res.data;
 
-		if (!geoJsonRes || !geoJsonRes['features'] || !geoJsonRes['features'][0] || !geoJsonRes['features'][0]['geometry']) {
-			Log.error('Error in geoJson:', geoJsonRes);
-			return;
-		}
-		Log.info('Gush Helka polygon is:', JSON.stringify(geoJsonRes['features'][0]['geometry']));
-		const geometryRes = geoJsonRes['features'][0]['geometry'];
-		if (geometryRes['type'] == 'Polygon') {
-			return geometryRes;
+		const features = (await Promise.all(
+			helkaArr.map(async (helka) => {
+				try {
+					const res = await axios(
+						{
+							url: `https://open.govmap.gov.il/geoserver/opendata/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=2.0.0&TYPENAMES=opendata:PARCEL_ALL&TYPENAME=opendata:PARCEL_ALL&outputFormat=json&cql_FILTER=GUSH_NUM=${filteredGush}AND%20PARCEL=${helka}&srsName=EPSG:4326`,
```

Comment:
I know the ugly peace of code is mine, but if you could extract the url to const I would appreciate it :)

Reply:
I can but then we need to parse it since we build the params dynamically